export.js

```diff
@@ -0,0 +1,161 @@
+function grabExcelValues(rows, artifact, objTemplate, customFieldRange) {
+    if (rows === null) {
+        getRows(artifact, objTemplate, customFieldRange);
+    }
+    else {
+        return Excel.run(function (context) {
+            let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+            let inputRange = columnRanges[artifact] + rows;
+            let customRange = (customFieldRange[0] + 3) + ":" + (customFieldRange[1] + rows);
+            let inputValues = sheet.getRange(inputRange);
+            let customFields = sheet.getRange(customRange);
+            inputValues.load();
+            customFields.load();
+            return context.sync()
+                .then(function () {
+                    customFieldObjArr = customFields.values.map(customFieldObjCreate);
+                    ajaxExport(inputValues.values, artifact, objTemplate, customFieldObjArr);
+                })
+        });
+    }
+}
+
+function getRows(artifact, objTemplate, customFieldRange) {
```

Comment:
Can you consider combining getRows with grabExcelValues differently. I like how you don't have a massive grabExcelValues function but you could streamline the code. Rather than getRows calling grabExcelValues, what if grabExcelValues used getRows to get the rows it needs?
in grabExcelValues set a new var `rowsToUse` that you use in the rest of the function. This also removes the need for an if else.
`var rowsToUse = rows || getRows(artifact);`

then you make a simpler getRows (code below should probably be split into a few lines. This function could also be more useful elsewhere - I don't think it is used outside of grabExcelValues currently so hopefully no side effects:
`function getRows(artifact) { return context.workbook.worksheets.getItem(toSheetName(artifact)).getUsedRange().values.length }`